device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/https/HttpsIotHubConnection.java

```diff
@@ -173,11 +173,9 @@ public IotHubStatusCode sendMessage(Message message) throws TransportException
                 //Codes_SRS_HTTPSIOTHUBCONNECTION_34_067: [If the response from the service is OK or OK_EMPTY, this function shall notify its listener that a message was sent with no exception.]
                 this.listener.onMessageSent(transportMessage, null);
             }
-            else
-            {
-                //Codes_SRS_HTTPSIOTHUBCONNECTION_34_068: [If the response from the service not OK or OK_EMPTY, this function shall notify its listener that a message was with the mapped IotHubServiceException.]
-                this.listener.onMessageSent(transportMessage, IotHubStatusCode.getConnectionStatusException(status, ""));
-            }
+
+            // Status codes other than 200 and 204 have their errors handled in the IotHubTransport layer once this method returns,
+            // so there is no need to call "this.listener.onMessageSent(transportMessage, someException)" from this layer.
```

Comment:
so was it throwing exception at both layers?

Reply:
Currently, the transport layer basically gets notified that the sent message failed twice. The first handling of the failed message works as expected, but the second handling fails because the message is already removed from the expected list and has already had its callback executed.